client/core/observeResource.go

```diff
@@ -64,7 +64,7 @@ func (d *Device) stopObservingResource(
 		return false, nil
 	}
 	d.observations.Delete(observationID)
-	o := v.(*observation)
+	o := v.(*observation) //nolint:forcetypeassert
```

Comment:
what about replace d.observations `sync.Map` with `gocoapSync.Map` ?

Reply:
Yeah, why not, good idea.